src/main/java/io/confluent/connect/jdbc/dialect/TimescaleDBDatabaseDialect.java

```diff
@@ -0,0 +1,98 @@
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.connect.jdbc.dialect;
+
+import io.confluent.connect.jdbc.dialect.DatabaseDialectProvider.SubprotocolBasedProvider;
+import io.confluent.connect.jdbc.sink.metadata.SinkRecordField;
+import io.confluent.connect.jdbc.util.*;
+import org.apache.kafka.common.config.AbstractConfig;
+
+import java.sql.*;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * A {@link DatabaseDialect} for TimescaleDB.
+ */
+public class TimescaleDBDatabaseDialect extends PostgreSqlDatabaseDialect {
+
+  /**
+   * The provider for {@link TimescaleDBDatabaseDialect}.
+   */
+  public static class Provider extends SubprotocolBasedProvider {
+    public Provider() {
+      super(TimescaleDBDatabaseDialect.class.getSimpleName(), "postgresql");
+    }
+
+    @Override
+    public DatabaseDialect create(AbstractConfig config) {
+      return new TimescaleDBDatabaseDialect(config);
+    }
+  }
+
+  static final int CHUNK_TIME_INTERVAL = 86400000;
+  static final String DELIMITER = ";";
+  static final String HYPERTABLE_WARNING = "A result was returned when none was expected";
+
+  /**
+   * Create a new dialect instance with the given connector configuration.
+   *
+   * @param config the connector configuration; may not be null
+   */
+  public TimescaleDBDatabaseDialect(AbstractConfig config) {
+    super(config);
+  }
+
+  @Override
+  public String buildCreateTableStatement(
+          TableId table,
+          Collection<SinkRecordField> fields
+  ) {
+    // This would create the table then convert it to a hyper table.
+    return super.buildCreateTableStatement(table, fields).concat(DELIMITER).concat(buildCreateHyperTableStatement(table));
+  }
+
+
+  public String buildCreateHyperTableStatement(
+          TableId table
+  ) {
+    ExpressionBuilder builder = expressionBuilder();
+
+    builder.append("SELECT create_hypertable('");
+    builder.append(table);
+    builder.append("', 'time', migrate_data => TRUE, chunk_time_interval => ");
+    builder.append(CHUNK_TIME_INTERVAL);
+    builder.append(");");
+    return builder.toString();
+  }
+
+  @Override
+  public void applyDdlStatements(
+          Connection connection,
+          List<String> statements
+  ) throws SQLException {
+    // This overrides the function by catching 'result was returned' error thrown by PSQL when creating hypertables
+    try{
+      super.applyDdlStatements(connection, statements);
```

Comment:
Problem here is that the applyDdlStatements as far as I know do not accept multiple statements in one query. So alternatively, do something like:
```java
List<String> updatedStatements = new ArrayList();
for (String statement : statements) {
  for (String substatement : statement.split(DELIMITER)) {
    updatedStatements.add(substatement);
  }
}
```
Only thing you'd have to watch out for is SQL injection, where a field has a `;` inside and it creates an additional query.

Safer and backwards-compatible would be to add a method to the generic Dialect:
```java
List<String> buildCreateTableStatements(TableId table,
           Collection<SinkRecordField> fields) {
// default implementation
Collections.singletonList(buildCreateTableStatement(table, fields))
}
```
and use that in the code that calls `applyDdlStatements`.

Reply:
Okay I see. I'll make the changes. Thanks!

Reply:
Yes, I've added the `DELIMITER` as a sort of hacky way to merge the two statements into one statement so that they were both executed in `applyDdlStatement`. I was trying to make changes to the least amount of files but I guess it's not very elegant and safe.  So I've made the changes to the `generic dialect` and `DbStructure`, as you suggested, instead. Thanks!